lib/drivers.js

```diff
@@ -0,0 +1,198 @@
+                      `readable. Specific error: ${err.message}`);
+    }
+
+    // if the file just doesn't exist, call it empty
+    return {};
+  }
+
+  const automationNames = [];
+  let msg = '';
+  for (const driverName of Object.keys(INSTALLED_DRIVERS)) {
+    const platformNames = INSTALLED_DRIVERS[driverName].platformNames;
+    if (!_.isArray(platformNames)) {
+      msg += `Entry for '${driverName}' has missing or ` +
+             `incorrect supported platformName list. `;
+    }
+
+    for (const pName of platformNames) {
+      if (!_.isString(pName)) {
+        msg += `Entry for '${driverName}' has an incorrectly ` +
+               `formatted platformName. `;
+      }
+    }
+
+    const automationName = INSTALLED_DRIVERS[driverName].automationName;
+    if (!_.isString(automationName)) {
+      msg += `Entry for '${driverName}' has missing ` +
+             `or incorrect automationName. `;
+    }
+    if (_.includes(automationNames, automationName)) {
+      msg += `Multiple drivers claim support for automationName ` +
+             `'${automationName}'. `;
+    }
+    automationNames.push(INSTALLED_DRIVERS[driverName].automationName);
+
+    if (!_.isString(INSTALLED_DRIVERS[driverName].version)) {
+      msg += `Entry for '${driverName}' has missing ` +
+             `or incorrect version. `;
+    }
+
+    if (!_.isString(INSTALLED_DRIVERS[driverName].pkgName)) {
+      msg += `Entry for '${driverName}' has missing ` +
+             `or incorrect NPM package name. `;
+    }
+
+    if (!_.isString(INSTALLED_DRIVERS[driverName].installSpec)) {
+      msg += `Entry for '${driverName}' has missing ` +
+             `or incorrect installation spec. `;
+    }
+
+    if (!_.isString(INSTALLED_DRIVERS[driverName].mainClass)) {
+      msg += `Entry for '${driverName}' has missing ` +
+             `or incorrect class name. `;
+    }
+
+    if (!_.includes(INSTALL_TYPES, INSTALLED_DRIVERS[driverName].installType)) {
+      msg += `Entry for '${driverName}' has missing ` +
+             `or incorrect install type. `;
+    }
+  }
+
+  if (msg) {
+    throw new Error(`Appium encountered an error while trying to verify ` +
+              `the data contained in the drivers file (${DRIVERS_JSON}): ` +
+              `${msg}Please ensure you have installed drivers correctly ` +
+              `using the Appium CLI.`);
+  }
+
+  return INSTALLED_DRIVERS;
+}
+
+async function writeInstalledDriver (name, data) {
+  INSTALLED_DRIVERS[name] = data;
+  await fs.writeFile(DRIVERS_JSON, JSON.stringify(INSTALLED_DRIVERS));
+}
+
+function printAvailableDrivers () {
+  const driverNames = Object.keys(INSTALLED_DRIVERS);
+  if (_.isEmpty(driverNames)) {
+    log.info('No drivers have been installed. Use the "appium driver" ' +
+             'command to install the one(s) you want to use.');
+    return;
+  }
+
+  log.info('Available drivers:');
+  for (const driverName of Object.keys(INSTALLED_DRIVERS)) {
```

Comment:
why not deconstruct directly here?

Reply:
`Object.keys(INSTALLED_DRIVERS)` is an array, not an object.

Reply:
I mean you can iterate `_.toPairs()` and then deconstruct it to `[driverName, {version, automationName}]`